packages/rum-core/src/domain/resource/matchRequestTiming.ts

```diff
@@ -32,18 +36,24 @@ export function matchRequestTiming(request: RequestCompleteEvent) {
   }
 
   const candidates = sameNameEntries
-    .map((entry) => entry.toJSON() as RumPerformanceResourceTiming)
-    .filter(toValidEntry)
+    .filter((entry) => !matchedResourceTimingEntries.has(entry))
+    .map((entry) => ({
+      original: entry,
+      serialized: entry.toJSON() as RumPerformanceResourceTiming,
+    }))
+    .filter((entry) => isValidEntry(entry.serialized))
     .filter((entry) =>
       isBetween(
-        entry,
+        entry.serialized,
         request.startClocks.relative,
         endTime({ startTime: request.startClocks.relative, duration: request.duration })
       )
     )
 
   if (candidates.length === 1) {
-    return candidates[0]
+    matchedResourceTimingEntries.set(candidates[0].original, PLACEHOLDER)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** any evidence that when two entries are matching the same timing, one entry has already matched another timing?

Reply:
In the case of fetch, having two timing matching the same resource without being already matched, means that both timings are available before the `then` of the first fetch is called.

This is not the case when doing `Promise.all()` but I don't exclude the possibility of it happening. In that case no timing will be matched.

Reply:
Would it be worth doing more experiments on browser behaviours around that? or measure the effectiveness of this change somehow?

Reply:
I'm adding a e2e test, however I've noticed that safari doesn't do like everyone and both `PerformanceResourceTiming` might be available at the time the first resource is matching it's timings.

This means that this PR reduce the amount of occurrence of missing resource timing on concurrent event (mainly chrome/FF) but does not prevent it totally (safari)

anyone has a better idea?

Reply:
as discussed during parking lot, this PR won't solve everything but is improving in some cases. So I've skipped the e2e on non chromium browsers as it's not reliable there.